### ISSUE TITLE
PR #5689 broke driver tests

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#5780]
https://github.com/flutter/flutter/pull/5689

Might be caused by the new version of `package:test`.

[build history](https://flutter-dashboard.appspot.com/build.html), [log](https://flutter-dashboard.appspot.com/api/get-log?ownerKey=ahNzfmZsdXR0ZXItZGFzaGJvYXJkclgLEglDaGVja2xpc3QiOGZsdXR0ZXIvZmx1dHRlci9kMmZkYTY3N2UxYjI5MTk1MzhkNmRmMTBlYmE4NTk4MDNiMTMxNDYyDAsSBFRhc2sYgICAgICAwAgM)

/cc @pq

**Comment by cbracken** [comment]
Similarly:
```shell
% flutter create --with-driver-test sample_app
% cd sample_app
% pub get
% flutter drive
```
Results in:
```
Building flutter tool...
Found connected emulator "iPhone 6s (Flutter)"; will reuse it.
Starting application: lib/main.dart
Observatory listening on http://127.0.0.1:8183
00:00 +0 -1: loading /Users/cbracken/src/sample_app/test_driver/main_test.dart
  Failed to load "/Users/cbracken/src/sample_app/test_driver/main_test.dart":
  Unable to spawn isolate: Unhandled exception:
  Could not import "package:test/src/runner/vm/catch_isolate_errors.dart" from "data:application/dart;charset=utf-8,%20%20%20%20%20%20%20%20import%20%22dart:isolate%22;%0A%0A%20%20%20%20%20%20%20%20import%20%22package:stream_channel/stream_channel.dart%22;%0A%0A%20%20%20%20%20%20%20%20import%20%22package:test/src/runner/plugin/remote_platform_helpers.dart%22;%0A%20%20%20%20%20%20%20%20import%20%22package:test/src/runner/vm/catch_isolate_errors.dart%22;%0A%0A%20%20%20%20%20%20%20%20import%20%22file:///Users/cbracken/src/sample_app/test_driver/main_test.dart%22%20as%20test;%0A%0A%20%20%20%20%20%20%20%20void%20main(_,%20SendPort%20message)%20%7B%0A%20%20%20%20%20%20%20%20%20%20var%20channel%20=%20serializeSuite(()%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20catchIsolateErrors();%0A%20%20%20%20%20%20%20%20%20%20%20%20return%20test.main;%0A%20%20%20%20%20%20%20%20%20%20%7D);%0A%20%20%20%20%20%20%20%20%20%20new%20IsolateChannel.connectSend(message).pipe(channel);%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20": Unsupported scheme used to locate .packages file: 'data:application/dart;charset=utf-8,%20%20%20%20%20%20%20%20import%20%22dart:isolate%22;%0A%0A%20%20%20%20%20%20%20%20import%20%22package:stream_channel/stream_channel.dart%22;%0A%0A%20%20%20%20%20%20%20%20import%20%22package:test/src/runner/plugin/remote_platform_helpers.dart%22;%0A%20%20%20%20%20%20%20%20import%20%22package:test/src/runner/vm/catch_isolate_errors.dart%22;%0A%0A%20%20%20%20%20%20%20%20import%20%22file:///Users/cbracken/src/sample_app/test_driver/main_test.dart%22%20as%20test;%0A%0A%20%20%20%20%20%20%20%20void%20main(_,%20SendPort%20message)%20%7B%0A%20%20%20%20%20%20%20%20%20%20var%20channel%20=%20serializeSuite(()%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20catchIsolateErrors();%0A%20%20%20%20%20%20%20%20%20%20%20%20return%20test.main;%0A%20%20%20%20%20%20%20%20%20%20%7D);%0A%20%20%20%20%20%20%20%20%20%20new%20IsolateChannel.connectSend(message).pipe(channel);%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20'.
```

**Comment by pq** [comment]
Suspicious indeed!  I'm looking a bit now...  If we don't make any quick progress may well be best to just revert.

**Comment by eseidelGoogle** [comment]
Flutter does not support data: urls.  If I recall correctly, @abarth was against adding support for such, but I may be mis-remembering.  That error would suggest that package:test now uses a data url to load a package?

**Comment by eseidelGoogle** [comment]
Possibly this commit?  https://github.com/dart-lang/test/commit/7ad2b1cad415f6af21755ca1e457f11d35255ba4

**Comment by abarth** [comment]
> If I recall correctly, @abarth was against adding support for such

I'm not sure what you're referring to.

**Comment by pq** [comment]
Interesting!

If that's the case, I wonder if backing up to ` 0.12.15+4` would work (if it's even possible)...

**Comment by eseidelGoogle** [comment]
@abarth I must be mistaken, then my apologies.

**Comment by cbracken** [comment]
Fixed by #5795. 

Long-term, the right solution here is to add an API for configurable .packages in package:test.

**Comment by pq** [comment]
Thanks for tracking this down Chris!
